pylxd/models/container.py

```diff
@@ -143,25 +143,52 @@ def get(self, filepath):
             return response.content
 
         def recursive_put(self, src, dst):
-            if os.path.isdir(src):
-                idx = len(os.path.dirname(src))
-                dst_dirs = []
-                for path, dirname, files in os.walk(src):
-                    dst_dir = os.path.join(dst, path[idx:].lstrip(os.path.sep))
-                    dst_dirs.append(dst_dir)
-
-                self._container.execute(['mkdir', '-p'] + dst_dirs)
-                # Copy files
+            """Recursively push directory to the container.
+
+            Recursively pushes directory to the containers
+            named by the `dst`
+
+            :param src: The source path of directory to copy.
+            :type filepath: str
+            :param dst: The destination path in the container
+                    of directory to copy
+            :type filepath: str
+            """
+            norm_src = os.path.normpath(src)
+            if not os.path.isdir(norm_src):
+                raise NotADirectoryError(
```

Comment:
This will fail on py2.7 -- `NotADirectoryError` only came into being with Py3.4 (IIRC!).

We either need to drop it to a `ValueError`or create a `NotADirectoryError` Exception class here, by subclassing `Exception` if we're in Python2.7.  Thoughts?